block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java

```diff
@@ -63,8 +65,12 @@ public final class PublisherHandler implements Pipeline<PublishStreamRequestUnpa
     private final long handlerId;
     /** The current streaming block number. This is used to track the current block being streamed. */
     private final AtomicLong currentStreamingBlockNumber;
-    /** The current block action. This is tracked to help the manager determine
-     * what to do with the current block. */
+    /** The unacknowledged yet blocks that were streamed to completion by this handler. */
+    private final ConcurrentSkipListSet<Long> unacknowledgedStreamedBlocks;
+    /**
+     * The current block action. This is tracked to help the manager determine
+     * what to do with the current block.
```

Comment:
Always prefer to have the first sentence end with a newline for Javadoc.
```suggestion
     * The current block action.
     * This is tracked to help the manager determine what to do with the current block.
```